packages/site/src/utils/theme.ts

```diff
@@ -0,0 +1,21 @@
+import { getLocalStorage, setLocalStorage } from './localStorage';
+
+/**
+ * Get the user's prefered theme in local storage.
+ * Will default to the browser's prefered theme if there is no value in local storage.
+ *
+ * @returns A string "light" or "dark" depending on the prefered theme.
+ */
+export const getThemePreference = () => {
+  const darkModeSystem = window?.matchMedia(
+    '(prefers-color-scheme: dark)',
+  ).matches;
+
+  const preference = getLocalStorage('theme');
+
+  if (!preference) {
+    setLocalStorage('theme', darkModeSystem ? 'dark' : 'light');
+  }
+
+  return preference === 'dark';
```

Comment:
We should probably use `darkModeSystem` if preference is undefined?

Reply:
Personal preference, I like the light mode more.

Reply:
Oh that's not what I meant. The current logic is flawed if no preference is currently stored it will not fall back to system until this function is called again